lib/livebook/session/data.ex

```diff
@@ -472,6 +486,26 @@ defmodule Livebook.Session.Data do
     |> unqueue_cells_evaluation(affected_cells_with_section)
   end
 
+  defp move_section({data, _} = data_actions, section, offset) do
+    updated_notebook = Notebook.move_section(data.notebook, section.id, offset)
+
+    cells_with_section_before = Notebook.elixir_cells_with_section(data.notebook)
+    cells_with_section_after = Notebook.elixir_cells_with_section(updated_notebook)
+
+    affected_cells_with_section =
+      cells_with_section_after
+      |> Enum.zip(cells_with_section_before)
+      |> Enum.drop_while(fn {{cell_before, _}, {cell_after, _}} ->
+        cell_before.id == cell_after.id
+      end)
+      |> Enum.map(fn {new, _old} -> new end)
+
+    data_actions
+    |> set!(notebook: updated_notebook)
+    |> mark_cells_as_stale(affected_cells_with_section)
+    |> unqueue_cells_evaluation(affected_cells_with_section)
```

Comment:
This is exactly the same se for the `move_cell` function above. To remove the duplication you could define `update_cells_status_after_moved(data_actions, prev_notebook)` and then you would just pipe into that function from both `move_cell` and `move_section`, like this:

```ex
updated_notebook = Notebook.move_section(data.notebook, section.id, offset)

data_actions
|> set!(notebook: updated_notebook)
|> update_cells_status_after_moved(data.notebook)
```